toolchains/BUILD.bazel

```diff
@@ -72,7 +72,6 @@ toolchain(
     name = "preinstalled_nmake_toolchain",
     exec_compatible_with = [
         "@platforms//os:windows",
-        "@bazel_tools//tools/cpp:msvc",
     ],
     toolchain = ":preinstalled_nmake",
     toolchain_type = ":make_toolchain",
```

Comment:
It feels like if its going to be a separate rule then its worth migrating this to a separate toolchain type specific to nmake? That would remove the need for the transition?

Reply:
ah but then you'd need a separate rule implementation instead of a macro but it should be able to reuse most of the existing `configure_make` rule?

Reply:
Sure, that would work. I assume @UebelAndre added  the `nmake` variant of the `make` toolchain as `nmake` and `make` are very similar, i.e. they have makefiles

Reply:
They're similar but this would also not require anyone to register extra toolchains. Thinking about the `cmake` rule, which takes a `make_toolchain`, I didn't think it should be necessary to have separate code paths for this variant of make. @jsharpe raised the question that maybe even `ninja` should be a `make_toolchain`. I disagree with this since I view `make` and `ninja` to be quite different. Though, from the perspective of `cmake`, they're not really. They're the underlying build tool that's used and it doesn't really care about much more than that. However, `rules_foreign_cc` supports more build systems than `cmake` as first class rules. So I feel we should view each build system as a unique rule with it's own toolchain. From what I've gathered though, `nmake` is just a variant of `gnu make`. There are some differences but is it really considered a separate build tool? To me it feels like a windows specific implementation of an existing tool. But if we feel that it's unique enough, then I'd say the correct path is to promote it to have it's own toolchain. Then yes, I'd say we should add rules for it which share the implementation of the existing `make` and `configure_make` rule.

Reply:
I'd be happy with either approach but of course you two are the `rules_foreign_cc` experts so I'll leave it up to you to decide and I'll implement what you decide.

Reply:
To help you guys decide, i have submitted PR #735 which implements the `configure_nmake` rule rather than a macro

Reply:
So, I think we should not write a separate rule for `nmake`. I think the `make` and `configure_make` rules should be able to work regardless of any make variant and for any variant we find difficult to fit into the existing rule, then perhaps it should be promoted to have it's own toolchain. But I don't think `nmake` is an example of that. So I'd vote against the other PR

Reply:
@jsharpe If you agree with @UebelAndre I will close PR #735 and address comments on this PR :)

Reply:
Yes I agree with @UebelAndre lets move forward with this approach.